plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -581,6 +581,9 @@ function OPDSBrowser:showDownloads(item)
     if item.author then
         filename = item.author .. " - " .. filename
     end
+    -- Ensure the default download filename is consistent across different platforms. In particular
+    -- this is needed for Progress Sync to work correctly if it is configured to rely on file names.
+    filename = filename:gsub('[\\,%/,:,%*,%?,%",%<,%>,%|]','_')
```

Comment:
Given the explanation in the PR I was expecting something more along the lines of exporting the `replaceAllInvalidChars` function and updating wherever `util.getSafeFilename()` is used to use that instead. Because now you're first doing this and still doing `util.getSafeFilename()` again unnecessarily later. Or just to add an extra boolean to `util.getSafeFilename()` for example.
https://github.com/koreader/koreader/blob/acbde33b2b91a41074b550cc25471f234b8c4b83/frontend/util.lua#L949-L960

But that aside, I suppose this same concern would apply just about anywhere the function is used, wouldn't it?